Mono.Cecil/AssemblyReader.cs

```diff
@@ -855,6 +855,12 @@ void InitializeTypeDefinitions ()
 
 				types [i] = ReadType (i + 1);
 			}
+
+			if (module.IsWindowsMetadata()) {
+				for (uint i = 0; i < length; i++) {
+					WindowsRuntimeProjections.Project (types [i]);
+				}
+			}
```

Comment:
After looking at the code some more, I think we should add the same "Project" call on the type that gets returned by "public TypeDefinition GetTypeDefinition (uint rid)" in this file.